### DISCUSSION TITLE
Why my PopupContextWindow behave odd?

### DISCUSSION THREAD

**Discussion by aeris170** [ocornut/imgui#3900] [Old/Archived discussions]
```
if (ImGui::CollapsingHeader("blabla", ImGuiTreeNodeFlags_DefaultOpen)) {
   //... stuff
}
if (ImGui::BeginPopupContextWindow(nullptr, ImGuiMouseButton_Right, false)) { 
   //... stuff
}
```

I'm using the above code to create a context popup that should only appear when I press and release RMB while my mouse is on the collapsing header. However, it also appears if I press RMB somewhere else, move my mouse on top of the collapsing header, and release RMB. Any idea why?

I didn't want to open an issue because I think the problem is on my end. If this is a bug, I'll open an issue.

**Comment by ocornut** [comment]
This is the behavior of popups under windows, they always react on RMB up and allow RMB down to close other popups.

**Reply by aeris170** [reply]
Hmm, so it is an OS-related issue. Is it possible to fix this? I tried setting a flag when RMB is pressed while hovering but I couldn't fix it.

**Reply by ocornut** [reply]
You are not specifying why this is an issue. I'm telling you this is the standardized way it works on regular Windows apps and this is how we model most behavior of Dear ImGui.

**Reply by aeris170** [reply]
You are right, sorry. 

Here is a video showing why this is an issue for me. Basically, I have a "render area" my app is rendering its contents to. I can control the "render area"'s camera via holding RMB. If I hold RMB, drag my mouse to the left and let go, the context menu of "Scene" window appears. I only want it to appear when I press down RMB while on it, not when I release RMB on it.

https://user-images.githubusercontent.com/25724155/110942133-6d546c80-834a-11eb-96d3-e2ac3908e31e.mp4

**Reply by ocornut** [reply]
I think the solution would involve:
- we could rework the Popup functions to make it possible to select if we can want click that have been used to grab active id (vs click that have been used to close popups)
- the Viewport would need to grab active id (via e.g. an invisible button)

Best to open an issue + include your video in it.

**Reply by aeris170** [reply]
Alright, what should I name the issue?

**Reply by ocornut** [reply]
"Popup functions not opening on mouse-up if mouse button activated another item?" maybe?

**Reply by aeris170** [reply]
Done! #3901
